warp/src/multipass/identity.rs

```diff
@@ -207,23 +225,49 @@ pub struct Identity {
     status_message: Option<String>,
 }
 
+#[cfg_attr(target_arch = "wasm32", wasm_bindgen::prelude::wasm_bindgen)]
 impl Identity {
     pub fn set_username(&mut self, user: &str) {
         self.username = user.to_string()
     }
 
+    pub fn set_status_message(&mut self, message: Option<String>) {
+        self.status_message = message
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    #[cfg_attr(target_arch = "wasm32", wasm_bindgen::prelude::wasm_bindgen(js_name = set_short_id))]
+    pub fn set_short_id_wasm(&mut self, id: String) {
+        self.short_id = id.try_into().unwrap();
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    #[cfg_attr(target_arch = "wasm32", wasm_bindgen::prelude::wasm_bindgen(js_name = set_did_key))]
+    pub fn set_did_key_wasm(&mut self, pubkey: String) {
+        use std::str::FromStr;
+        self.did_key = DID::from_str(pubkey.as_str()).unwrap();
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    #[cfg_attr(target_arch = "wasm32", wasm_bindgen::prelude::wasm_bindgen(js_name = set_created))]
+    pub fn set_created_wasm(&mut self, time: js_sys::Date) {
+        self.created = time.into();
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    #[cfg_attr(target_arch = "wasm32", wasm_bindgen::prelude::wasm_bindgen(js_name = set_modified))]
+    pub fn set_modified_wasm(&mut self, time: js_sys::Date) {
+        self.modified = time.into();
+    }
+}
+impl Identity {
     pub fn set_short_id<I: Into<ShortId>>(&mut self, id: I) {
         self.short_id = id.into()
     }
 
     pub fn set_did_key(&mut self, pubkey: DID) {
         self.did_key = pubkey
     }
-
-    pub fn set_status_message(&mut self, message: Option<String>) {
-        self.status_message = message
-    }
```

Comment:
Why is this function moved behind wasm cfg attr?

Reply:
Hmm, thought i saw a `#[cfg(target_arch = "wasm32")]` as well as `#[cfg_attr(target_arch = "wasm32", wasm_bindgen::prelude::wasm_bindgen)]` hence the question.